web/src/main/java/org/openmrs/web/attribute/handler/ProviderFieldGenDatatypeHandler.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.attribute.handler;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.Provider;
+import org.openmrs.api.context.Context;
+import org.openmrs.customdatatype.CustomDatatype;
+import org.openmrs.customdatatype.CustomDatatype.Summary;
+import org.openmrs.customdatatype.InvalidCustomValueException;
+import org.openmrs.customdatatype.datatype.ProviderDatatype;
+/**
+ * An implementation of {@link FieldGenDatatypeHandler} for {@link ProviderDatatype}
+ * @since 1.12
+ *
+ */

```

Comment:
This should extend BaseMetadataFieldGenDatatypeHandler and get rid of the methods here that are already implemented in the base class